### DISCUSSION TITLE
Cannot connect Shelly 2PM to wifi

### DISCUSSION THREAD

**Discussion by GijsTimmers** [arendst/Tasmota#20896] [Q&A]
I'm running `tasmota32solo1.bin` 12.2.0 on my ESP32 Shelly 2PM device.

For some reason, my Shelly will not connect to wifi.

After issuing `reset 2`, it will start an access point. I can connect to it, find my wifi network and enter the password.

From there, one of the following scenarios will happen:

- Tasmota will say that my credentials are wrong (I'm sure I entered them right)
- The device will say that it got the expected `192.168.3.xxx` IP, an entry will show up in the DHCP table, but when trying to go to the new IP address, Tasmota will for some reason disconnect from the wifi and start its own access point again with a server on 192.168.4.1;
- Sometimes I get the `Connect failed with AP timeout` ([just as in this ESP8266 issue](https://github.com/arendst/Tasmota/discussions/12512))
- In one instance, I was able to connect to the new `192.168.3.xxx` IP, but it disconnected soon after

Any help is appreciated

**Comment by Jason2866** [comment]
Which brand is your Router or/and Access Points?
Have you flashed Tasmota via OTA or serial (wired)?

**Comment by GijsTimmers** [comment]
It's a TP Link Acher C6 v2 flashed with OpenWrt

I flashed Tasmota via serial.

DHCP log in router:
```
Sat Mar  9 13:15:44 2024 daemon.info dnsmasq-dhcp[1]: DHCPOFFER(br-iot) 192.168.3.207 redacted-mac-address
Sat Mar  9 13:15:44 2024 daemon.info dnsmasq-dhcp[1]: DHCPDISCOVER(br-iot) redacted-mac-address
Sat Mar  9 13:15:44 2024 daemon.info dnsmasq-dhcp[1]: DHCPOFFER(br-iot) 192.168.3.207 redacted-mac-address
Sat Mar  9 13:15:44 2024 daemon.info dnsmasq-dhcp[1]: DHCPDISCOVER(br-iot) redacted-mac-address
Sat Mar  9 13:15:44 2024 daemon.info dnsmasq-dhcp[1]: DHCPOFFER(br-iot) 192.168.3.207 redacted-mac-address
Sat Mar  9 13:15:44 2024 daemon.info dnsmasq-dhcp[1]: DHCPDISCOVER(br-iot) redacted-mac-address
Sat Mar  9 13:15:44 2024 daemon.info dnsmasq-dhcp[1]: DHCPOFFER(br-iot) 192.168.3.207 redacted-mac-address
Sat Mar  9 13:15:45 2024 daemon.info dnsmasq-dhcp[1]: DHCPREQUEST(br-iot) 192.168.3.207 redacted-mac-address
Sat Mar  9 13:15:45 2024 daemon.info dnsmasq-dhcp[1]: DHCPACK(br-iot) 192.168.3.207 redacted-mac-address tasmota-669BD0-7120
```

**Reply by Jason2866** [reply]
Suggest to start from scratch. Erase whole flash. Flash latest version via Tasmota Web Installer
https://tasmota.github.io/install/

**Reply by GijsTimmers** [reply]
OK, I'll try.
